client/lib/signup/step-actions/index.js

```diff
@@ -608,12 +636,13 @@ function processItemCart(
 	}
 }
 
-function prepareItemForAddingToCart( item ) {
+function prepareItemForAddingToCart( item, lastKnownFlow = null ) {
 	return {
 		...item,
 		extra: {
 			...item.extra,
 			context: 'signup',
+			...( lastKnownFlow && { signup_flow: lastKnownFlow } ),
```

Comment:
Even though this file is not TypeScript, could we add `signup_flow` to the TS definition of [RequestCartProductExtra](https://github.com/Automattic/wp-calypso/blob/689f747bb416bd05fa81c914f288cc3268774f32/packages/shopping-cart/src/types.ts#L553)? Some day we may filter these properties and I want to make sure all the ones we use are documented.

Reply:
should be there now